src/App.tsx

```diff
@@ -5,7 +5,16 @@ import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const addMovie = (movie: Movie) => {
+    const findImdbId = movies
+      .find(oldMovie => oldMovie.imdbId === movie.imdbId);
+
+    if (!findImdbId) {
+      setMovies([...movies, movie]);
```

Comment:
If the new state is computed use the previous state, you can pass a function to `setMovies`.